### DISCUSSION TITLE
使用 setState 方法，直接报"Maximum call stack size exceeded"，新手卡了很久快被劝退了。

### DISCUSSION THREAD

**Discussion by dongdong7** [NervJS/taro#13628] [Q & A] [answered]
### 相关平台
H5

**浏览器版本: Google Chrome  版本 111.0.5563.146**
**使用框架: React**

### 复现步骤

 所有用到  setState 这个方法就报 Maximum call stack size exceeded
 
 函数如下： 

  onSave = (e) => {
    this.setState({

    });
  };

 html如下： 

     <View className="but-box">
          <View className="button">
            <Button
              color="#0462CA"
              block
              className="submit"
              loading={submit_loading}
              disabled={submit_disabled}
              onClick={this.onSave.bind()}
            >
              保 存
            </Button>
          </View>
        </View>

 之前用taro做过一两个项目从没碰到过这个问题，望大神帮我看一下

### 期望结果
正常运行 setState 方法

### 实际结果
调用 setState 就报 Maximum call stack size exceeded

### 环境信息
```
👽 Taro v3.6.4


  Taro CLI 3.6.4 environment info:
    System:
      OS: Linux 5.15 Ubuntu 22.04.1 LTS 22.04.1 LTS (Jammy Jellyfish)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 14.21.3 - ~/.nvm/versions/node/v14.21.3/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v14.21.3/bin/yarn
      npm: 6.14.18 - ~/.nvm/versions/node/v14.21.3/bin/npm
    npmPackages:
      @tarojs/cli: 3.6.4 => 3.6.4 
      @tarojs/components: 3.6.4 => 3.6.4 
      @tarojs/helper: 3.6.4 => 3.6.4 
      @tarojs/plugin-framework-react: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-alipay: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-h5: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-jd: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-qq: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-swan: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-tt: 3.6.4 => 3.6.4 
      @tarojs/plugin-platform-weapp: 3.6.4 => 3.6.4 
      @tarojs/react: 3.6.4 => 3.6.4 
      @tarojs/runtime: 3.6.4 => 3.6.4 
      @tarojs/shared: 3.6.4 => 3.6.4 
      @tarojs/taro: 3.6.4 => 3.6.4 
      @tarojs/webpack5-runner: 3.6.4 => 3.6.4 
      babel-preset-taro: 3.6.4 => 3.6.4 
      eslint-config-taro: 3.6.4 => 3.6.4 
      react: ^18.2.0 => 18.2.0 

```

**Comment by dongdong7** [comment]
<img width="377" alt="3" src="https://user-images.githubusercontent.com/230118/230720023-de6dbda7-2d55-4afe-8dcd-20df37a37803.png">

报错内容如上。

**Comment by TheKonka** [comment]
简单试了一下，没有复现，有空可以上传个demo仓库 
https://github.com/NervJS/taro/issues/13593

**Comment by smithyj** [comment, marked as answer]
你立即执行了，用箭头函数

**Comment by smithyj** [comment]
或者使用autobind-decoration

**Comment by smithyj** [comment]
@TheKonka 有群吗大佬，想进开发群啊
